### PR TITLE
Issue #53: About Mac Activity.

### PR DESCRIPTION
**Pull request by aayushsoni10**
about mac activity

You need to follow the below instructions carefully

0. You are not supposed to send PR of your forked repo's `master` branch
1. An issue that your PR is fixing should exist in the Issue section
2. Issue should have the "approved" label
3. In the first line in the description of your PR, include issue number as the first message. 
  eg. 
  `Issue #123`
4. Commit message should follow the following regexp patter
`^Issue #\d+: .$`
It must **not** end with a period, space, tab or a newline
valid egs. `Issue #123: This is my complete commit message`
invalid egs. 
`Issue #123: This is my complete commit message.` or
`This is my complete commit message where I forgot Issue reference` or
`Issue #123 Forgot colon in commit message` or
`Issue 123: This is my complete commit message but forgot hash symbol`

5. There is no restriction on PR title format

Thanks for reading! You can remove this whole message now and type in your description.

### CODE REVIEW

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        sharedPreferences = getSharedPreferences(AHC.USER_PREFERENCES, Context.MODE_PRIVATE);
+        text = sharedPreferences.getString(AHC.TEXT_ABOUT_MAC, "");
+        url = sharedPreferences.getString(AHC.IMAGEURL_ABOUT_MAC, "");
+        if ("".equals(text)) {
+            textView.setText(TIAB1 + TIAB2 + TIAB3 + TIAB4 + TIAB5 + TIAB6);
```
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: Is there an issue when you replace them with a single string?
Were you facing reduced coverage when using R.strings?
Or some checkstyle warning when holding the value in one string?

I still think using R.strings is the better way as this should not be the job a Helper Class

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        ButterKnife.bind(this);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        sharedPreferences = getSharedPreferences(AHC.USER_PREFERENCES, Context.MODE_PRIVATE);
+        text = sharedPreferences.getString(AHC.TEXT_ABOUT_MAC, "");
```
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: What we can do here is use R.string for default value

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    @BindView(R.id.toolbar_about_mac)
+    Toolbar toolbar;
+
+    /**
+     * Database Reference.
+     */
+    private DatabaseReference databaseReference;
+
+    /**
+     * Vale event listener.
```
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: minor typo

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+ * @version 1.0.
+ */
+
+public class AboutMACActivity extends AppCompatActivity {
+
+    /**
+     * About Mac Text.
+     */
+    @BindView(R.id.about_mac_text)
+    TextView textView;
```
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: make variable name more descriptive

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+    /**
+     * About Mac Image.
+     */
+    @BindView(R.id.about_mac_image)
+    ImageView imageView;
+
+    /**
+     * About Mac Toolbar.
```
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: similar for this imageview,

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java (hunk 2 above)
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: learn how to get string object from R.string

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+                        .into(imageView);
+                final SharedPreferences.Editor editor = sharedPreferences.edit();
+                editor.putString(AHC.TEXT_ABOUT_MAC, text);
+                editor.putString(AHC.IMAGEURL_ABOUT_MAC, url);
+                editor.apply();
+            }
+
+            @Override
+            public void onCancelled(final DatabaseError databaseError) {
+                Log.e(TAG, "ERROR");
```
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: The error message is extremely vague.
check for a method in the `databaseError` obj that gives more details
Also use the printStackTrace method

File: app/src/main/java/in/ac/bits_pilani/goa/ard/utils/AHC.java [hunk 7]
```diff
@@ -58,4 +58,61 @@
...
      */
     public static final String FDR_NAV_DRAWER_IMAGE_LIST = "navDrawerImages";
+
+    /**
+     * Firebase Database Reference for aboutMAC.
+     */
+    public static final String FDR_ABOUT_MAC = "aboutMAC";
+
+    /**
+     * String shtml.
```
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/utils/AHC.java]: If this is a key, please mention so

File: app/src/main/java/in/ac/bits_pilani/goa/ard/utils/AHC.java [hunk 8]
```diff
@@ -58,4 +58,61 @@
...
+     */
+    public static final String TEXT_ABOUT_MAC = "text";
+
+    /**
+     * String surl.
```
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/utils/AHC.java]: same here

File: app/src/main/java/in/ac/bits_pilani/goa/ard/utils/AHC.java [hunk 9]
```diff
@@ -58,4 +58,61 @@
...
+     */
+    public static final String IMAGEURL_ABOUT_MAC = "imageUrl";
+
+    /**
+     * Float magicnumber.
```
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/utils/AHC.java]: please mention in the comments, what this magic number is used for...
and avoid naming it so if possible

File: app/src/main/java/in/ac/bits_pilani/goa/ard/utils/AHC.java [hunk 10]
```diff
@@ -58,4 +58,61 @@
...
+    /**
+     * Sharedpreferences name for ARD APP.
+     */
+    public static final String USER_PREFERENCES = "ARD_SHP";
+
+    /**
+     * String 1 text content for About mac textview.
+     * TIAB = TEXT INITIAL ABOUT MAC.
+     */
+    public static final String TIAB1 = "Mobile App Club - BITS Goa is an Android developers club";
```
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/utils/AHC.java]: See other comment

File: app/src/main/res/layout/content_about_mac.xml [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+    <ImageView
+        android:contentDescription="@string/imageview_about_mac_description"
+        android:id="@+id/about_mac_image"
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/about_mac_image_height"
+        android:scaleType="centerCrop"
+        android:src="@drawable/mac_logo"/>
+
+    <TextView
+        android:id="@+id/about_mac_text"
```
Comment (kukreja-vikramaditya) [inline app/src/main/res/layout/content_about_mac.xml]: about_mac_desc

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java (hunk 1 above)
Comment (aayushsoni10) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: checkstyle
as 120 character is maximum in a line of code.

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java (hunk 2 above)
Comment (aayushsoni10) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: for ""
or 
for AHC.variables.

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java (hunk 1 above)
Comment (aayushsoni10) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: Stored this in string.

File: app/src/main/java/in/ac/bits_pilani/goa/ard/utils/AHC.java (hunk 10 above)
Comment (aayushsoni10) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/utils/AHC.java]: removed this

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java (hunk 1 above)
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: Try R.string and for coverage use Mockito to mock the sharedpreferences file

If at the end it still doesn't work out we can add a suppression in checkstyle (But I think it is possible by method 1)

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java (hunk 2 above)
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: For ""

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java (hunk 1 above)
Comment (aayushsoni10) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: R.string is working

File: app/src/main/java/in/ac/bits_pilani/goa/ard/utils/AHC.java (hunk 9 above)
Comment (aayushsoni10) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/utils/AHC.java]: GLIDE_THUMBNAIL

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+    protected void onStart() {
+        super.onStart();
+        sharedPreferences = getSharedPreferences(AHC.USER_PREFERENCES, Context.MODE_PRIVATE);
+        text = sharedPreferences.getString(AHC.TEXT_ABOUT_MAC, getString(R.string.empty_about_mac));
+        url = sharedPreferences.getString(AHC.IMAGEURL_ABOUT_MAC, getString(R.string.empty_about_mac));
+        if (getString(R.string.empty_about_mac).equals(text)) {
+            textView_about_mac.setText(getString(R.string.initial_text_about_mac));
+        } else {
+            textView_about_mac.setText(text);
+            Glide.with(getApplicationContext()).load(url)
```
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: Can we have some sort of fade animation here? The change in image is very abrupt

File: app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+                    .into(imageView_about_mac);
+        }
+        databaseReference = FirebaseDatabase.getInstance().getReference().child(AHC.FDR_ABOUT_MAC);
+        listener = databaseReference.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(final DataSnapshot dataSnapshot) {
+                text = dataSnapshot.child(AHC.TEXT_ABOUT_MAC).getValue(String.class);
+                url = dataSnapshot.child(AHC.IMAGEURL_ABOUT_MAC).getValue(String.class);
+                textView_about_mac.setText(text);
+                Glide.with(getApplicationContext()).load(url)
```
Comment (kukreja-vikramaditya) [inline app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java]: Same here

**Pull request closed without merging**